### PR TITLE
Deploy Win32Binaries in a separate folder within a UWP project

### PR DESCRIPTION
**Pull request by rido-min**
This PR updates Step 1, 2, 3 and 4 of the JourneySamples.

It uses one UWP project to perform Appx configuration and packaging, and one Debuging project using (using the VSIX extension) to debug the win32 project in the appx context.

To avoid known bugs in the .NET native compiler, all win32 binaries are stored in a separate folder.

### CODE REVIEW

File: Samples/JourneyAcrossTheBridge/DesktopBridgeDemo - Step1/MyDesktopApp.Package/package.appxmanifest [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10"
+  xmlns:mp="http://schemas.microsoft.com/appx/2014/phone/manifest"
+  xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10"
+  xmlns:rescap="http://schemas.microsoft.com/appx/manifest/foundation/windows10/restrictedcapabilities" 
+  IgnorableNamespaces="uap mp rescap">
+
+  <Identity
+    Name="MyDesktopApp-Step1"
+    Version="1.0.0.0"
+    Publisher="CN=rmpablos" />
```
Comment (vladimp) [inline Samples/JourneyAcrossTheBridge/DesktopBridgeDemo - Step1/MyDesktopApp.Package/package.appxmanifest]: Please change to - 
Publisher="CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US"

File: Samples/JourneyAcrossTheBridge/DesktopBridgeDemo - Step2/MyDesktopApp.Package/package.appxmanifest [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10"
+  xmlns:mp="http://schemas.microsoft.com/appx/2014/phone/manifest"
+  xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10"
+  xmlns:rescap="http://schemas.microsoft.com/appx/manifest/foundation/windows10/restrictedcapabilities" 
+  IgnorableNamespaces="uap mp rescap">
+
+  <Identity
+    Name="MyDesktopApp-Step2"
+    Version="1.0.0.0"
+    Publisher="CN=rmpablos" />
```
Comment (vladimp) [inline Samples/JourneyAcrossTheBridge/DesktopBridgeDemo - Step2/MyDesktopApp.Package/package.appxmanifest]: Publisher="CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US"

File: Samples/JourneyAcrossTheBridge/DesktopBridgeDemo - Step3/MyDesktopApp.Package/BundleArtifacts/x64.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+MainPackage=C:\code\DesktopBridgeToUWP-Samples\Samples\JourneyAcrossTheBridge\DesktopBridgeDemo - Step3\MyDesktopApp.Package\bin\x64\Release\MyDesktopApp.Package_1.0.4.0_x64.appx
```
Comment (vladimp) [inline Samples/JourneyAcrossTheBridge/DesktopBridgeDemo - Step3/MyDesktopApp.Package/BundleArtifacts/x64.txt]: This file and below shouldn't be committed right?

File: Samples/JourneyAcrossTheBridge/DesktopBridgeDemo - Step3/MyDesktopApp.Package/MyDesktopApp.Package.jsproj [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup Condition="'$(VisualStudioVersion)' == '' or '$(VisualStudioVersion)' &lt; '15.0'">
+    <VisualStudioVersion>15.0</VisualStudioVersion>
+  </PropertyGroup>
+  <Import Project="$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\$(WMSJSProjectDirectory)\Microsoft.VisualStudio.$(WMSJSProject).Default.props" />
+  <Import Project="$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\$(WMSJSProjectDirectory)\Microsoft.VisualStudio.$(WMSJSProject).props" />
+  <PropertyGroup>
+    <TargetPlatformIdentifier>UAP</TargetPlatformIdentifier>
+    <TargetPlatformVersion>10.0.14393.0</TargetPlatformVersion>
+    <TargetPlatformMinVersion>10.0.10586.0</TargetPlatformMinVersion>
```
Comment (vladimp) [inline Samples/JourneyAcrossTheBridge/DesktopBridgeDemo - Step3/MyDesktopApp.Package/MyDesktopApp.Package.jsproj]: 14393

File: Samples/JourneyAcrossTheBridge/DesktopBridgeDemo - Step3/MyDesktopApp.AppxPackage/App.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+
+namespace MyDesktopApp.AppxPackage
```
Comment (vladimp) [inline Samples/JourneyAcrossTheBridge/DesktopBridgeDemo - Step3/MyDesktopApp.AppxPackage/App.xaml.cs]: This file should be in step 4, why is it in step 3?

File: Samples/JourneyAcrossTheBridge/DesktopBridgeDemo - Step1/MyDesktopApp.AppxDebug/MyDesktopApp.AppxDebug.vsdproj [hunk 6]
```diff
@@ -1,16 +1,19 @@
...
     <ProjectTypeGuids>{365BE41E-7993-430D-A5E6-1381A54BC077};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
-    <MinimumVisualStudioVersion>14</MinimumVisualStudioVersion>
+    <ProjectGuid>{8B192FE7-1E7C-4C4F-9E07-DDA8375B0F7B}</ProjectGuid>
+    <MinimumVisualStudioVersion>15</MinimumVisualStudioVersion>
+    <DisableFastUpToDateCheck>true</DisableFastUpToDateCheck>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|x86'">
     <DefineConstants>DEBUG</DefineConstants>
     <OutputPath>bin\x86\Debug\</OutputPath>
+    <PackageLayout>..\MyDesktopApp.AppxPackage\bin\x86\Debug</PackageLayout>
```
Comment (vladimp) [inline Samples/JourneyAcrossTheBridge/DesktopBridgeDemo - Step1/MyDesktopApp.AppxDebug/MyDesktopApp.AppxDebug.vsdproj]: Please consider to use environment variables for x86 and debug/release

File: Samples/JourneyAcrossTheBridge/DesktopBridgeDemo - Step1/MyDesktopApp.AppxDebug/MyDesktopApp.AppxDebug.vsdproj [hunk 7]
```diff
@@ -1,16 +1,19 @@
...
   <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
   <PropertyGroup>
     <Configuration Condition="'$(Configuration)' == ''">Debug</Configuration>
     <Platform Condition="'$(Platform)' == ''">x86</Platform>
-    <ProjectGuid>{31BA40E5-C0FC-47F6-BB7A-A9091082484B}</ProjectGuid>
     <ProjectTypeGuids>{365BE41E-7993-430D-A5E6-1381A54BC077};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
-    <MinimumVisualStudioVersion>14</MinimumVisualStudioVersion>
+    <ProjectGuid>{8B192FE7-1E7C-4C4F-9E07-DDA8375B0F7B}</ProjectGuid>
+    <MinimumVisualStudioVersion>15</MinimumVisualStudioVersion>
+    <DisableFastUpToDateCheck>true</DisableFastUpToDateCheck>
```
Comment (vladimp) [inline Samples/JourneyAcrossTheBridge/DesktopBridgeDemo - Step1/MyDesktopApp.AppxDebug/MyDesktopApp.AppxDebug.vsdproj]: Why we need this - DisableFastUpToDateCheck?

File: Samples/JourneyAcrossTheBridge/DesktopBridgeDemo - Step1/MyDesktopApp.AppxDebug/MyDesktopApp.AppxDebug.vsdproj (hunk 6 above)
Comment (ridomin) [inline Samples/JourneyAcrossTheBridge/DesktopBridgeDemo - Step1/MyDesktopApp.AppxDebug/MyDesktopApp.AppxDebug.vsdproj]: this is aligned on how other msbuild properties are set. Inside the condition we set the platform/configuration constants.

**Pull request merged**